### PR TITLE
Add Connect Gateway as a requirement for Managed Control Plane

### PR DESCRIPTION
**Pull request by ruigulala**

### CODE REVIEW

**Comment by djazayeri** [comment]
lgtm, but @zerobfd should approve.

**Comment by ruigulala** [comment]
Adjusted based on the new plan. PTAL. @djazayeri @zerobfd

File: asmcli/lib/installation_dependencies.sh [hunk 1]
```diff
@@ -185,6 +191,10 @@ enable_gcloud_apis(){
   info "Enabling required APIs..."
   # shellcheck disable=SC2046
   retry 3 gcloud services enable --project="${PROJECT_ID}" $(required_apis | tr '\n' ' ')
+  if is_managed; then
+    # shellcheck disable=SC2046
+    retry 3 gcloud services enable --project="${PROJECT_ID}" $(old_managed_required_apis | tr '\n' ' ')
```
Comment (djazayeri) [inline asmcli/lib/installation_dependencies.sh:196]: I thought we didn't need to explicitly enable this, since it's auto-enabled by `mesh`.

Comment (ruigulala) [reply asmcli/lib/installation_dependencies.sh]: Thanks for the catch! Removed.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**